dtfs-central-api/src/v1/controllers/tfm/amendments/tfm-get-amendments.controller.ts

```diff
@@ -72,7 +72,7 @@ export const getAmendmentsByFacilityId = async (req: Request, res: Response) =>
       default:
         if (amendmentIdOrStatus) {
           if (!ObjectId.isValid(amendmentIdOrStatus)) {
-            return res.status(400).send({ status: 400, message: 'Invalid amendment Id' });
+            return res.status(400).send({ status: 400, message: 'Invalid amendment Id', code: API_ERROR_CODE.INVALID_MONGO_ID_PATH_PARAMETER });
```

Comment:
could we throw an error here instead? the ApiError check exists below anyway

Reply:
I don't think so. This particular case is actually an annoying one for me because the reason this check needs to exist here is that the path parameter can be either a mongo id or an amendment status, so I can't use the middleware like I have everywhere else. If I wanted to throw here, I would need to create a new error class for invalid path parameters which I don't want to do as I want to discourage validating path parameters in the controller and encourage using the middleware

Reply:
Interesting -- I agree with the discouraging, unsure how it'll work in practice as this is still a common pattern regardless of middleware handling in the legacy code.

`amendmentIdOrStatus` is certainly troublesome!

Sorry I just like being part of these discussions